pkg/apply/format.go

```diff
@@ -93,6 +93,68 @@ func FormatSettingsDiff(
 	return string(bytes.TrimRight(buf.Bytes(), "\n")), nil
 }
 
+// FormatSettingsDiffTracker formats the settings diffs as a ChangesTracker object instead of a table
+func FormatSettingsDiffTracker(
+	topicName string,
+	topicSettings config.TopicSettings,
+	configMap map[string]string,
+	diffKeys []string,
+) (*ChangesTracker, error) {
+
+	configEntries := make([]ConfigEntryChanges, 0)
+	for _, diffKey := range diffKeys {
+		configValueStr := configMap[diffKey]
+
+		var valueStr string
+		var err error
+
+		if topicSettings.HasKey(diffKey) {
+			valueStr, err = topicSettings.GetValueStr(diffKey)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		configEntries = append(configEntries, ConfigEntryChanges{
+			Name:    diffKey,
+			Current: configValueStr,
+			Updated: valueStr,
+		})
+	}
+
+	return &ChangesTracker{
+		Topic:              topicName,
+		NumPartitions:      IntValueChanges{},
+		ReplicationFactor:  IntValueChanges{},
+		ReplicaAssignments: []ReplicaAssignmentChanges{},
+		ConfigEntries:      configEntries,
+		MissingKeys:        []string{},
+		Action:             ActionEnumUpdate,
+	}, nil
+}
+
+// processes TopicConfig object from topic creation into a ChangesTracker
+func ProcessTopicConfigIntoChanges(topicName string, topicConfig kafka.TopicConfig) *ChangesTracker {
+	configEntries := make([]ConfigEntryChanges, 0)
+	for _, entry := range topicConfig.ConfigEntries {
+		configEntries = append(configEntries, ConfigEntryChanges{
+			Name:    entry.ConfigName,
+			Current: entry.ConfigValue,
```

Comment:
IF the topic is new `current` should not be set. Can you set it to nil ?

Reply:
unfortunately you can't set a value in a struct to `nil` in go. I'll try splitting `ChangesTracker` out into `NewChangesTracker` and `UpdateChagesTracker`